reg/drone/srv/air_data_computer/_.0.1.uavcan

```diff
@@ -0,0 +1,48 @@
+# A generic air data computer (ADC) service.
+#
+# An air data computer service is generally a non-interactive publish-only service: when activated,
+# it keeps publishing its measurements at a fixed rate over several subjects until deactivated.
+# The activation/deactivation, if supported, is managed via the standard arming command service.
+# The data update rates and other parameters are controlled via the Register API using implementation-defined
```

Comment:
We need to standardize this in some way on the service level (in DS-015). Otherwise it generated ambiguity and divergence.

Reply:
Making the right decision on what to standardize is at least as important as deciding what *not* to standardize. Nano-management and overspecialization by the standard may harm the ecosystem just as badly as the lack of clear standards; the recent story with OpenCL v3 appears to be an interesting illustration of that.

Our objective here is to prepare a set of building blocks that can be mixed and matched by the integrator to create distributed systems, where it is up to the integrator to configure the involved elements correctly. The auto-configuration is something that we have on the roadmap past the MVP milestone, and it may or may not involve defining standard register names, but at any rate, right now it is unnecessary and premature.

The v1 of DS-015 will require editing many registers per each involved node, but this is something that we are already agreed on, as we were discussing it for some time already and we already have scheduled the work on the CLI register editing tool to be done in the near future.

Reply:
> The v1 of DS-015 will require editing many registers per each involved node, but this is something that we are already agreed on, as we were discussing it for some time already and we already have scheduled the work on the CLI register editing tool to be done in the near future.

I think there's a miscommunication here: one thing is that you are certain of which registers need to live in a specific device, which has specific functions associated with it, and the the thing that you edit it's the values of the registers, not the register name by itself (also you don't want to manually created more registers, because it's clear what are the functions that device has). The other is to actually allow the edition of everything, which while still possible, it doesn't facilitate the work of the integrator - having to define everything is a burden and should be avoided.

I am not saying this is something that needs to exist for v1 of DS-015. What I am saying is that we need to consider that most of these things should be easy enough to integrate and expand in a way that are easy to adopt and use. Otherwise, no one is going to bother.

Reply:
Is there any action item specific to this pull request that you would like to propose?

Reply:
I think we need to make sure we standardize register naming that correspond to things that are used basically in all nodes for most of the topics, like the update rate, and also the registers for pub/sub port IDs. This will make sure that we have a strong convention that guarantees that the integrator knows what registers to look for when it wants to change the register values.

So for this particular PR, I would start by defining what registers should by default exist on a node with a particular function and its naming. Other registers can exist, but as you specify already, those can be implementation-defined.

Reply:
There may be multiple network service instances per node, so I don't think it's possible to define specific register names. We could define wildcards as has been proposed for `uavcan.sub.*`, for example. But notice that for a human integrator it doesn't make much difference whether, say, the rate of a particular sensor feed is set via a spec-adherent register or some implementation-defined register. Where it matters is the auto-configuration, but this capability is expressly excluded from the scope of DS-015 v1.0. My plan is to leave this undefined for now, and get back to it after we have the MVP out.

Reply:
> We could define wildcards as has been proposed for uavcan.sub.*, for example

Let's do that as an intermediate step here then, please.

Reply:
Not possible without redefining the project goals and disrupting the schedule. The plan is to release the standard in October; defining the register names would require exploring the auto-configuration options, it is major work and it will take time.

Reply:
> defining the register names would require exploring the auto-configuration options

Why is that? What does it take to define wildcards on the data types definitions that can define the way that the registers are named? I don't see where exactly does that impact on the project and where does that imply auto-configuration options. We are just stating that a human integrator will know exactly what would be the wildcard to look for when configuring a register. That for me is quite clear.

Reply:
> What does it take to define wildcards on the data types definitions that can define the way that the registers are named?

It depends on what our final intent is.

If the intent is to define some arbitrary requirements, it can be done in a minute.

If the intent is to define a self-consistent and viable standard, it takes work. The register name patterns are an inherent part of the auto-configuration capability, and it will be impossible to define them without constructing the general framework of the auto-configuration service first.

What you are requesting is impossible to do under the current schedule and budget.